creator-node/src/snapbackSM/peerSetManager.js

```diff
@@ -0,0 +1,221 @@
+const axios = require('axios')
+
+const { logger } = require('../logging')
+
+// TODO: config var
+const PEER_HEALTH_CHECK_REQUEST_TIMEOUT = 2000
+
+class PeerSetManager {
+  constructor ({ discoveryProviderEndpoint, creatorNodeEndpoint }) {
+    this.discoveryProviderEndpoint = discoveryProviderEndpoint
+    this.creatorNodeEndpoint = creatorNodeEndpoint
+  }
+
+  log (msg) {
+    logger.info(`SnapbackSM:::PeerSetManager: ${msg}`)
+  }
+
+  logError (msg) {
+    logger.error(`SnapbackSM:::PeerSetManager ERROR: ${msg}`)
+  }
+
+  /**
+   * Retrieves node users, paritions it into a slice, computes the peer set of the node users,
+   * and performs a health check on the peer set.
+   * @param {Object[]} nodeUsers array of objects of schema { primary, secondary1, secondary2, user_id, wallet }
+   * @returns the unhealthy peers in a Set
+   *
+   * @note consider returning healthy set?
+   */
+  async getUnhealthyPeers (nodeUsers) {
+    // Compute content node peerset from nodeUsers (all nodes that are in a shared replica set with this node)
+    let peerSet = this.computeContentNodePeerSet(nodeUsers)
```

Comment:
what if this call fails? previous code has it try-catched, not sure why you removed?

Reply:
if this line fails, then the outer try/catch block will catch the error. the decision tree will not have an explicit message regarding `getUnhealthyPeers()` failing, but any potential error will be thrown and caught. 

I left this as a comment earlier on this PR -- we shouldn't need these nested try/catch blocks in general. these nested try/catch blocks add unnecessary complexity in this flow, especially if the pure purpose of these nested calls are to add to the decision tree. this can be accomplished via a custom error handler class, or phase tracking as we do in the signup flow.

Reply:
yeah as i mentioned above, the purpose was to get more custom error messages, but you're right i think the way i did it is pretty messy and verbose

anyway sounds good